apiary-metastore-events/sns-metastore-events/apiary-metastore-listener/src/main/java/com/expediagroup/apiary/extensions/events/metastore/listener/ApiarySnsListener.java

```diff
@@ -134,7 +134,21 @@ public void onInsert(InsertEvent event) throws MetaException {
     if (event.getStatus() == false) {
       return;
     }
-    publishInsertEvent(EventType.INSERT, event.getDb(), event.getTable(), event.getPartitionKeyValues(),
+
+    Table table = event.getTableObj();
+    String databaseName = table.getDbName();
+    String tableName = table.getTableName();
+
+    LinkedHashMap<String, String> partitionKeyValues =  new LinkedHashMap<>();
```

Comment:
Can the type be `Map` on the LHS?

Reply:
I'm not sure to be honest - this was taken directly from Hive 2. I'm assuming there had to be some reasoning or internal use case to hive that relies on maintaining order in the map

Reply:
Yes. I think Adrian you mentioned this in an old PR for this project. IIRC, I tested this and LinkedHashMap on LHS is important to preserve order.

Reply:
Well, I could see why it being of type `LinkedHashMap` is important but wouldn't think using the interface for the LHS would break anything. Where we do this in https://github.com/ExpediaGroup/apiary-extensions/pull/67/files#diff-4a1373cd1422a872232ed9c786f7f307R45 the LHS type is `Map`.

Reply:
Ah. Yes. you are right. We left the LHS as LinkedHashMap coz we wanted to explicitly tell someone reading the code that order matters in this case. I found my PR where you made this comment :)

https://github.com/ExpediaGroup/apiary-extensions/pull/16#discussion_r226689852

Reply:
Oh, snap! 😆

Reply:
made the change then reverted from the comment 😆.
Current state is LinkedHashMap on LHS